src/Jobs/Notifications/CouponNotificationJob.php

```diff
@@ -0,0 +1,184 @@
+<?php
+declare(strict_types=1);
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Jobs\Notifications;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\ActionScheduler\ActionSchedulerInterface;
+use Automattic\WooCommerce\GoogleListingsAndAds\Coupon\CouponHelper;
+use Automattic\WooCommerce\GoogleListingsAndAds\Exception\InvalidValue;
+use Automattic\WooCommerce\GoogleListingsAndAds\Google\NotificationsService;
+use Automattic\WooCommerce\GoogleListingsAndAds\Jobs\AbstractActionSchedulerJob;
+use Automattic\WooCommerce\GoogleListingsAndAds\Jobs\ActionSchedulerJobMonitor;
+use Automattic\WooCommerce\GoogleListingsAndAds\Jobs\JobInterface;
+use Automattic\WooCommerce\GoogleListingsAndAds\Value\NotificationStatus;
+
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class CouponNotificationJob
+ * Class for all the Coupons Notifications Jobs
+ *
+ * @since x.x.x
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Jobs\Notifications
+ */
+class CouponNotificationJob extends AbstractActionSchedulerJob implements JobInterface {
+
+	/**
+	 * @var NotificationsService $notifications_service
+	 */
+	protected $notifications_service;
+
+	/**
+	 * @var CouponHelper $coupon_helper
+	 */
+	protected $coupon_helper;
+
+	/**
+	 * Notifications Jobs constructor.
+	 *
+	 * @param ActionSchedulerInterface  $action_scheduler
+	 * @param ActionSchedulerJobMonitor $monitor
+	 * @param NotificationsService      $notifications_service
+	 * @param CouponHelper              $coupon_helper
+	 */
+	public function __construct(
+		ActionSchedulerInterface $action_scheduler,
+		ActionSchedulerJobMonitor $monitor,
+		NotificationsService $notifications_service,
+		CouponHelper $coupon_helper
+	) {
+		$this->notifications_service = $notifications_service;
+		$this->coupon_helper         = $coupon_helper;
+		parent::__construct( $action_scheduler, $monitor );
+	}
+
+	/**
+	 * Get the job name
+	 *
+	 * @return string
+	 */
+	public function get_name(): string {
+		return 'notifications/coupons';
+	}
+
+
+	/**
+	 * Logic when processing the items
+	 *
+	 * @param array $args Arguments with the item id and the topic
+	 */
+	protected function process_items( array $args ): void {
+		if ( ! isset( $args[0] ) || ! isset( $args[1] ) ) {
+			do_action(
+				'woocommerce_gla_error',
+				'Error sending Coupon Notification. Topic and Coupon ID are mandatory',
+				__METHOD__
+			);
+			return;
+		}
+
+		$item  = $args[0];
+		$topic = $args[1];
+
+		if ( $this->can_process( $item, $topic ) && $this->notifications_service->notify( $topic, $item ) ) {
+			$this->set_status( $item, $this->get_after_notification_status( $topic ) );
+			$this->maybe_mark_as_unsynced( $topic, $item );
+		}
```

Comment:
If we're not able to process the item it would be nice to throw an Exception with some basic information so that the scheduled action doesn't fail silently.

Reply:
Do you mean to throw exceptions each time AS cannot process?  Wouldn't this cause excessive filling of the logs? For example when we have the case mentioned in `https://github.com/woocommerce/google-listings-and-ads/pull/2278#issuecomment-1968846588`

Reply:
My thought was that it would make things more traceable if in the Action Scheduler logs we had a bit of additional information instead of just seeing that the action ran but if you think it would be excessive it's definitely not an issue to leave it as is.